app/scripts/models/account.js

```diff
@@ -713,24 +713,43 @@ define(function (require, exports, module) {
     /**
      * Fetch the account's device list and populate the `devices` collection.
      *
-     * @param {object} devices - Devices collection
+     * @param {object} clients - Devices collection
      * @returns {promise} - resolves when complete
      */
-    fetchDevices: function (devices) {
-      var sessionToken = this.get('sessionToken');
+    fetchDevices: function (clients) {
+      return this._fxaClient.deviceList(this.get('sessionToken'))
+        .then((devices) => {
+          devices.map((item) => {
+            item.clientType = Constants.CLIENT_TYPE_DEVICE;
+          });
+
+          return clients.add(devices);
+        });
+    },
+
+    /**
+     * Fetch the OAuth Apps for the given account, populated the passed in
+     * oAuthApps collection.
+     *
+     * @param {Object} clients - Clients collection used to store list.
+     * @returns {Promise} resolves when the action completes
+     */
+    fetchOAuthApps: function (clients) {
+      return this._oAuthClient.fetchClients(this.get('accessToken'))
```

Comment:
The `.fetchClients` name here works, but wonder if `.fetchApps` is more appropriate and might lessen confusion since clients are OAuth apps and devices. Also, this change might be out of scope of this PR.